src/Rings/mpoly-local.jl

```diff
@@ -68,25 +88,66 @@ elem_type(::MPolyRingLoc{T}) where {T} = MPolyElemLoc{T}
 elem_type(::Type{MPolyRingLoc{T}}) where {T} = MPolyElemLoc{T}
 parent_type(::Type{MPolyElemLoc{T}}) where {T} = MPolyRingLoc{T}
 
+function check_parent(a::MPolyElemLoc{T}, b::MPolyElemLoc{T}, thr::Bool = true) where {T}
+  if parent(a) == parent(b)
+    return true
+  elseif thr
+     error("Parent rings do not match")
+  else
+    return false
+  end
+end
+
+
 ###############################################################################
 # Arithmetics                                                                 #
 ###############################################################################
 
-(W::MPolyRingLoc)() = MPolyElemLoc(W.base_ring(), W.max_ideal)
-(W::MPolyRingLoc)(i::Int) = MPolyElemLoc(W.base_ring(i), W.max_ideal)
-(W::MPolyRingLoc)(i::RingElem) = MPolyElemLoc(W.base_ring(i), W.max_ideal)
-(W::MPolyRingLoc)(f::MPolyElem) = MPolyElemLoc(f, W.max_ideal)
-(W::MPolyRingLoc)(g::AbstractAlgebra.Generic.Frac) = MPolyElemLoc(g, W.max_ideal)
+(W::MPolyRingLoc)() = W(W.base_ring())
+(W::MPolyRingLoc)(i::Int) = W(W.base_ring(i))
+(W::MPolyRingLoc)(i::RingElem) = W(W.base_ring(i))
+
+function (W::MPolyRingLoc{T})(f::MPolyElem) where {T}
+  return MPolyElemLoc{T}(f//one(parent(f)), W)
+end
+
+function (W::MPolyRingLoc{T})(g::AbstractAlgebra.Generic.Frac) where {T}
+  return MPolyElemLoc{T}(g, W)
+end
+
 (W::MPolyRingLoc)(g::MPolyElemLoc) = W(g.frac)
-Base.one(W::MPolyRingLoc) = MPolyElemLoc(one(W.base_ring), W.max_ideal)
-Base.zero(W::MPolyRingLoc) = MPolyElemLoc(zero(W.base_ring), W.max_ideal)
+Base.one(W::MPolyRingLoc) = W(1)
+Base.zero(W::MPolyRingLoc) = W(0)
+
+# Since a.parent.max_ideal is maximal and AA's frac arithmetic is reasonable,
+# none of these ring operations should generate bad denominators.
+# If this turns out to be a problem, remove the last false argument.
+function +(a::MPolyElemLoc{T}, b::MPolyElemLoc{T}) where {T}
+  check_parent(a, b)
+  return MPolyElemLoc{T}(a.frac + b.frac, a.parent, false)
+end
```

Comment:
Could also do this (slightly shorter, but otherwise should behave identically ... I think...(
```suggestion
function +(a::T, b::) where T <: MPolyElemLoc
  check_parent(a, b)
  return T(a.frac + b.frac, a.parent, false)
end
```

Reply:
Also noted.